### PR TITLE
[Ego-19] Demo Config Flag to get Demo User on Login

### PR DESCRIPTION
**Pull request by joneubank**
Demo server will allow all user's to log in and no user data to be persisted.

This is a repeat of work done in #20 but updated to work with the JPA refactor.

Enums are added back in, UserStatus now includes all 4 values permittable by the DB schema.

https://github.com/overture-stack/ego/issues/19

### CODE REVIEW

File: src/main/java/org/overture/ego/model/enums/UserRole.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+public enum UserRole {
+  USER("USER"),
+  ADMIN("ADMIN");
+
+  @NonNull private final String value;
+
+  @Override
+  public String toString() {
+    return value;
+  }
```
Comment (andricDu) [inline src/main/java/org/overture/ego/model/enums/UserRole.java]: I like a new line between closing brace of last method and closing brace of class/enum

File: src/main/java/org/overture/ego/model/enums/UserStatus.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  DISABLED("Disabled"),
+  PENDING("Pending"),
+  REJECTED("Rejected"),;
+
+  @NonNull private final String value;
+
+  @Override
+  public String toString() {
+    return value;
+  }
```
Comment (andricDu) [inline src/main/java/org/overture/ego/model/enums/UserStatus.java]: same thing again for a new line. Just a style thing.

File: src/main/java/org/overture/ego/service/UserService.java [hunk 3]
```diff
@@ -20,30 +20,42 @@
...
 import org.springframework.data.domain.Page;
 import org.springframework.data.domain.Pageable;
 import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
 import org.springframework.util.StringUtils;
 
 import java.text.SimpleDateFormat;
 import java.util.Date;
 import java.util.List;
+import java.util.Optional;
```
Comment (andricDu) [inline src/main/java/org/overture/ego/service/UserService.java]: unused import, codacy confirms.

**Comment by joneubank** [comment]
Comments addressed.

**Pull request merged**